libs/execution_context/ebpf_program.c

```diff
@@ -31,16 +31,16 @@ typedef struct _ebpf_program
         struct ubpf_vm* vm;
     } code_or_vm;
 
-    ebpf_extension_client_t* global_helper_extension_client;
-    ebpf_extension_data_t* global_helper_provider_data;
-    ebpf_extension_dispatch_table_t* global_helper_provider_dispatch_table;
+    ebpf_extension_client_t* generic_helper_extension_client;
+    ebpf_extension_data_t* generic_helper_provider_data;
+    ebpf_extension_dispatch_table_t* generic_helper_provider_dispatch_table;
 
     ebpf_map_t** maps;
     size_t count_of_maps;
 
     ebpf_extension_client_t* program_info_client;
     const void* program_info_binding_context;
-    const ebpf_extension_data_t* program_info_data;
+    const ebpf_extension_data_t* program_info_provider_data;
```

Comment:
The windows API naming convention is to not duplicate the containing structure suffix into prefixes of the member fields of that structure.  Since the containing structure is ebpf_program, it implies every field is implicitly prefixed with "program", just like it appears in line 100 etc.   So "program->info_client" would be better (less redundant) than "program->program_info_client".

Reply:
Would you prefer  "info_provider_data"?
The name of the NPI is "program info", and program happens to have pointers to the client and provider_data of this NPI.